src/gnark_backend_glue/mod.rs

```diff
@@ -72,7 +73,22 @@ pub fn prove_with_pk(
     values: Vec<FieldElement>,
     proving_key: &[u8],
 ) -> Result<Vec<u8>> {
-    todo!()
+    let rawr1cs = RawR1CS::new(*circuit, values)?;
+
+    // Serialize to json and then convert to GoString
+    let rawr1cs_json = serde_json::to_string(&rawr1cs)?;
+    let rawr1cs_c_str = CString::new(rawr1cs_json)?;
+    let rawr1cs_go_string = GoString::from_cstring(&rawr1cs_c_str);
+
+    let proving_key_serialized = std::str::from_utf8(proving_key)?.to_string();
```

Comment:
We could do this instead
```suggestion
    let proving_key_serialized = String::from_utf8(proving_key)?;
```